azurelinuxagent/ga/exthandlers.py

```diff
@@ -304,10 +319,52 @@ def cleanup_outdated_handlers(self):
             if os.path.isfile(pkg):
                 try:
                     os.remove(pkg)
-                    logger.verbose("Removed extension package {0}".format(pkg))
-                except OSError as e:
-                    logger.warn("Failed to remove extension package {0}: {1}".format(pkg, e.strerror))
-   
+                    logger.verbose("Removed extension package "
+                                "{0}".format(pkg))
+                except Exception as e:
+                    logger.warn("Failed to remove extension package: "
+                                "{0}".format(pkg))
+    
+    #TODO: Why is this here with extension handlers... this doesn't seem right.
+    def handle_remote_access(self):
+        logger.verbose("Entered handle_remote_access")
+        self.protocol.client.update_goal_state(True)
+        self.protocol.client.update_remote_access_conf(self.protocol.client.goal_state)
+        remote_access = self.protocol.client.remote_access
+        if remote_access is not None:
+            # Get existing users.
+            osUtils = get_osutil()
+            existingUsers = osUtils.getusers()
+            userNames = []
+            for usr in existingUsers:
+                userNames.append(usr[0])            
+            for acc in remote_access.Users:
+                try:
+                    dateTimeString = acc.Expiration
+                    accExp = datetime.strptime(dateTimeString, "%a, %d %b %Y %H:%M:%S %Z") + timedelta(days=1)
+                    now = datetime.utcnow()
+                    if acc.Name not in userNames and now < accExp :
+                        logger.verbose("Adding user {0} with expiration {1}".format(acc.Name, acc.Expiration))
+                        expirationString = accExp.strftime("%Y-%m-%d")
+                        osUtils.useradd(acc.Name, expirationString)
+                        cachefile = os.path.join(conf.get_lib_dir(), "temp.dat")
+                        prvKey = os.path.join(conf.get_lib_dir(), "TransportPrivate.pem")
+                        pwd = self.cryptUtil.decryptSecret(acc.EncryptedPassword, prvKey, cachefile)   
+                        osUtils.chpasswd(acc.Name, pwd, conf.get_password_cryptid(), conf.get_password_crypt_salt_len())
+                        osUtils.conf_sudoer(acc.Name)
+                        logger.info("User '{0}' added successfully".format(acc.Name))
+                except Exception as e:
+                    #TODO: Better error handling and cap to retry logic.
+                    logger.error("handle_remote_access: {0}".format(str(e)))
+        else:
+            logger.verbose("handle_remote_access remote_access is null")
+    
+    def addUser(self):
+        pass
+    
+    def removeUser(self):
```

Comment:
If you are not going to implement why define them?

Reply:
just placeholders.  You said I should submit the PR before the code was ready so this was submitted only partially done.  In the final iteration these will be either filled out completely or removed.